tsk3.c

```diff
@@ -510,7 +510,7 @@ static Directory File_as_directory(File self) {
         RaiseError(EInvalidParameter, "Invalid parameter: self->info.");
         return NULL;
     }
-    if(self->info->meta == NULL || self->info->meta->type != TSK_FS_META_TYPE_DIR) {
+    if(self->info->meta == NULL || !(TSK_FS_IS_DIR_META(self->info->meta->type))) {
```

Comment:
Would be checking for `TSK_FS_META_TYPE_VIRT_DIR` and `TSK_FS_META_TYPE_DIR` be sufficient?

This probably needs a libtsk version check as well.

Reply:
The `TSK_FS_IS_DIR_META` macro does exactly this :
https://github.com/sleuthkit/sleuthkit/blob/develop/tsk/fs/tsk_fs.h#L403

Reply:
Ack but it was introduced in a specific version of TSK https://github.com/sleuthkit/sleuthkit/blame/develop/tsk/fs/tsk_fs.h#L403

please add version guards